### DISCUSSION TITLE
How to create a common function to call multiple useFetch

### DISCUSSION THREAD

**Discussion by mogi86** [nuxt/nuxt#16143] [Questions]
Hi. I want to create a common function to call multiple useFetch. Is there any good way to do this?
I want to use a common function with multiple page components. I know I can use multiple useFetch in `<script setup>`.
However, when I call multiple useFetches in a function in a cut out external file, I get `nuxt instance unavailable` error. (Reproduction code)

In the actual code, `Promise.all` cannot be used because the result of the first useFetch is used as a parameter for the second useFetch.

Reproduction: https://github.com/mogi86/study-nuxt3-v2/blob/main/src/pages/multiple.vue

**Comment by IsraelOrtuno** [comment]
You can't call multiple async methods in a function handler (see https://github.com/nuxt/framework/issues/8844).
Also this can help: https://github.com/nuxt/framework/issues/5740#issuecomment-1229197529

**Reply by mogi86** [reply]
Thanks.
I understand I can't create a common function. So, it's impossible to create common logic, not just functions. Is it correct?
